pymongo/encryption.py

```diff
@@ -552,6 +554,95 @@ def __init__(
 
+    def create_encrypted_collection(
+        self,
+        database: Database,
+        name: str,
+        kms_provider: Optional[str] = None,
+        data_key_opts: Optional[Mapping[str, Any]] = None,
+        **kwargs: Any,
+    ) -> Tuple[Collection[_DocumentType], Mapping[str, Any]]:
+        """Create a collection with encryptedFields.
+
+        Normally collection creation is automatic. This method should
+        only be used to specify options on
+        creation. :class:`~pymongo.errors.CollectionInvalid` will be
+        raised if the collection already exists.
+
+        :Parameters:
+          - `name`: the name of the collection to create
+          - `kms_provider`: the KMS provider to be used
+          - `data_key_opts` (dict): a dictionary containing additional arguments to the `create_data_key` method that will be
+            used to auto-generate the data keys if `keyId` is `None` such as::
+
+                {
+                   masterKey: Optional<Document>
+                   keyAltNames: Optional<Array[String]>
+                   keyMaterial: Optional<BinData>
+                }
+
+          - ``encryptedFields`` (optional) (dict): **(BETA)** Document that describes the encrypted fields for
+            Queryable Encryption. For example::
+
+                {
+                  "escCollection": "enxcol_.encryptedCollection.esc",
+                  "eccCollection": "enxcol_.encryptedCollection.ecc",
+                  "ecocCollection": "enxcol_.encryptedCollection.ecoc",
+                  "fields": [
+                      {
+                          "path": "firstName",
+                          "keyId": Binary.from_uuid(UUID('00000000-0000-0000-0000-000000000000')),
+                          "bsonType": "string",
+                          "queries": {"queryType": "equality"}
+                      },
+                      {
+                          "path": "ssn",
+                          "keyId": Binary.from_uuid(UUID('04104104-1041-0410-4104-104104104104')),
+                          "bsonType": "string"
+                      }
+                    ]
+                }
+
+            The "keyId" may be set to `None` to auto-generate the data keys.
+          - `**kwargs` (optional): additional keyword arguments are the same as "create_collection".
+
+        All optional `create collection command`_ parameters should be passed
+        as keyword arguments to this method.
+        See the documentation for :meth:`~pymongo.database.Database.create_collection` for all valid options.
+
+        .. versionadded:: 4.4
+           Added the codec_options, read_preference, and write_concern options.
+
+        .. _create collection command:
+            https://mongodb.com/docs/manual/reference/command/create
+
+        """
+        if not data_key_opts:
+            data_key_opts = {}
+        encrypted_fields = database._get_encrypted_fields(kwargs, name, False)
```

Comment:
https://github.com/mongodb/specifications/pull/1376 proposes not checking AutoEncryptionOptions in createEncryptedCollection. This may need to be updated when that is merged.

Reply:
Thanks for the heads-up!

Reply:
The referenced PR has been merged

Reply:
I have already committed my fix for this.

Reply:
I think you also need to add the associated docstring clarification.

Reply:
The docstring now shows encryption_fields as a required argument and I removed the note in the error message relating to this.

Reply:
I was referring to: "Drivers MUST document that `createEncryptedCollection` does not affect any
auto encryption settings on existing MongoClients that are already configured with 
auto encryption.  Users must configure auto encryption after creating the 
encrypted collection with the `createEncryptedCollection` helper."